sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs

```diff
@@ -182,6 +182,43 @@ public BlobClient(Uri blobUri, TokenCredential credential, BlobClientOptions opt
         }
         #endregion ctors
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlobClient"/>
+        /// class with an identical <see cref="Uri"/> source but the specified
+        /// <paramref name="snapshot"/> timestamp.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/creating-a-snapshot-of-a-blob" />.
+        /// </summary>
+        /// <param name="snapshot">The snapshot identifier.</param>
+        /// <returns>A new <see cref="BlobClient"/> instance.</returns>
+        /// <remarks>
+        /// Pass null or empty string to remove the snapshot returning a URL
+        /// to the base blob.
+        /// </remarks>
+        public new BlobClient WithSnapshot(string snapshot)
+        {
+            BlobUriBuilder blobUriBuilder = new BlobUriBuilder(Uri) { Snapshot = snapshot };
+            return new BlobClient(blobUriBuilder.ToUri(), Pipeline, Version, ClientDiagnostics, CustomerProvidedKey, ClientSideEncryption, EncryptionScope);
```

Comment:
Why do we unescape the path in `BlobBaseClient` but not here in `BlobClient`?  If we should be doing it everywhere, is that work fixing automatically as part of `BlobUriBuilder`'s .ctor?

Reply:
That was an oversight on my part.  `BlobUriBuilder.BlobName` and `.Path` should now be escaped, based on another PR I merged.  I'll address now.